gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/GemFireUpdateResultSet.java

```diff
@@ -91,6 +94,13 @@ public final void setNumRowsModified(int numRowsModified) {
 
   @Override
   protected final void openCore() throws StandardException {
+    TXStateInterface tx = this.gfContainer.getActiveTXState(this.tran);
+    if (tx == null && (this.gfContainer.isRowBuffer() || Misc.getGemFireCache().snapshotEnabledForTest())) {
+      // TOOD: decide on autocommit or this flag: Discuss
+      this.tran.getTransactionManager().begin(IsolationLevel.SNAPSHOT, null);
+      ((GemFireTransaction)lcc.getTransactionExecute()).setActiveTXState(TXManagerImpl.getCurrentTXState(), false);
+      this.tran.setImplicitSnapshotTxStarted(true);
+    }
```

Comment:
Hmm.. That was in anticipation that there may be sql update/delete directly fired on row buffer without iterating on column store.
If there is no such case, then it can be removed. It is harmless for column table because as per design txState at this point will always be not null if coming after iteration.